scripts/nice-history/nice-history.js

```diff
@@ -0,0 +1,220 @@
+// ==UserScript==
+// @name         Nice history
+// @match        https://pokefarm.com/*
+// @icon         https://pokefarm.com/favicon.ico
+// @grant        none
+// ==/UserScript==
+
+/*
+Author: Mirzam
+Keeps track of Nices recieved, so you can easily see who sent them and when!
+Nices are ONLY stored locally per device, since the main intent is to be able to view a missed nice.
+Report bugs and issues in my main code sharing thread: https://pfq.link/~-HvB
+
+For testing: Manually generate a test nice
+Test nices cannot be dismissed normally, but should be detected by the script
+
+username = 'USERNAME';
+testNice = document.createElement('div');
+testNice.className = 'nice_toast';
+testNice.innerHTML = '<div class="nice_star1"></div><div class="nice_star2"></div><div class="nice_star3"></div><a href="#" class="nice_close"></a>'+username+'<br>gave you a Nice!';
+document.body.append(testNice);
+
+*/
+
+(function() {
+  let userID = null;
+  let recentNices = [];
+
+  // You should always call this instead of accessing userID directly
+  const getUserID = () => {
```

Comment:
This is also provided by PFQ as `$.USERID`, which doesn't depend on a particular way of it being passed to JS-land in the first place.